pkg/frontend/querymiddleware/split_by_instant_interval.go

```diff
@@ -134,10 +137,18 @@ func (s *splitInstantQueryByIntervalMiddleware) Do(ctx context.Context, req Requ
 		return s.next.Do(ctx, req)
 	}
 
-	if stats.GetSplitQueries() == 0 {
+	if stats.GetSplitQueries() <= 0 {
 		// the query cannot be split, so continue
 		level.Debug(spanLog).Log("msg", "input query resulted in a no operation, falling back to try executing without splitting")
-		s.metrics.splittingSkipped.WithLabelValues(skippedReasonNoop).Inc()
+		if stats.GetNoOpSmallIntervalQuery() {
```

Comment:
Each query should only be tracked onces in the skipped, otherwise the sum of skipped + success doesn't match the attempted ones.